File: 0206-reverse-linked-list/0206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
  if (head === null) return null;
  let currentNode = head;
  const nodeArray = [];

  while (currentNode !== null) {
    nodeArray.push(currentNode);
    currentNode = currentNode.next;
  }

  for (let i = nodeArray.length - 1; i >= 0; i--) {
    nodeArray[i].next = nodeArray[i - 1] ?? null;
  }

  return nodeArray[nodeArray.length - 1];
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return None
        
        prev = head
        reverse = None
        temp = ListNode(head.val, head.next)

        while prev:
            temp = temp.next
            prev.next = reverse
            reverse = prev
            prev = temp
        
        return reverse
